ambari-server/src/main/java/org/apache/ambari/server/controller/AddServiceRequest.java

```diff
@@ -118,7 +118,7 @@ private AddServiceRequest(
     Configuration configuration
   ) {
     this.operationType = null != operationType ? operationType : OperationType.ADD_SERVICE;
-    this.recommendationStrategy = null != recommendationStrategy ? recommendationStrategy : ConfigRecommendationStrategy.NEVER_APPLY;
+    this.recommendationStrategy = null != recommendationStrategy ? recommendationStrategy : ConfigRecommendationStrategy.getDefault();
```

Comment:
I think its great to have a getDefault method here, but I'm concerned about using the same default as a Blueprints deployment. 

When the StackAdvisor integration was introduced in earlier versions of Ambari Blueprints, we decided to use "NEVER_APPLY" as the default strategy, in order to preserve backwards compatibility for existing users of Blueprints. 

For the "Add Service" API, I think we should consider a different default, perhaps "ALWAYS_APPLY_DONT_OVERRIDE_CUSTOM_VALUES" as the default for the Add Service implementation, but keep "NEVER_APPLY" as the default for Blueprints.